### PR TITLE
DEV: Add `pluginId` to squash warning from `pluginApi`

### PR DESCRIPTION
**Pull request by keegangeorge**
Started this to remove annoying warnings in console by plugin api:

```
To prevent errors, add a `pluginId` key to your changes when calling `modifyClass`
```

### CODE REVIEW

File: assets/javascripts/discourse/lib/layouts.js.es6 [hunk 1]
```diff
@@ -222,7 +224,7 @@ function setupContext(context, app) {
 
     if (controllerExists) {
       api.modifyClass(controllerClass, Sidebars);
```
Comment (keegangeorge) [inline assets/javascripts/discourse/lib/layouts.js.es6]: `api.modifyClass(controllerClass, Sidebars);`

Unsure of where to add here id for this line..

**Comment by angusmcleod** [comment]
@keegangeorge I actually haven't looked at this ``pluginId`` biz closely yet. @merefield would you mind reviewing this one and explaining how the new ``pluginId`` system works for keegan and I?

File: assets/javascripts/discourse/lib/layouts.js.es6 (hunk 1 above)
Comment (keegangeorge) [reply assets/javascripts/discourse/lib/layouts.js.es6]: @merefield I'm still unsure how to add the `pluginId` here. In this `modifyClass()` it is calling the sidebar mixin. I'm not sure how to add the pluginId to this line.

**Comment by merefield** [comment]
@keegangeorge my associated warnings have gone down from 45 to 14 with this PR, so progress there, can we kill the last 14?

**Comment by keegangeorge** [comment]
Hey @merefield,

The last commit (https://github.com/paviliondev/discourse-layouts/pull/18/commits/58331dd9e70874e436e09aad554b774afb330e3a) I added here seems to have gotten rid of the errors but I don't feel its the right approach.

Basically, everything else I tried for:
```
api.modifyClass(controllerClass, Sidebars);
``` 
it doesn't seem to be detecting the pluginId property when using a mixin and the 14 warnings become present saying the pluginId is required.

What I did here was just avoid having the modifyClass method called here, but I assume theres a better way to do it... Any thoughts on how we can have the modifyClass method pick up the pluginId property in the mixin?

**Pull request merged**